src/MediaCollections/Filesystem.php

```diff
@@ -77,6 +77,11 @@ public function copyToMediaLibraryFromRemote(RemoteFile $file, Media $media, ?st
         );
     }
 
+    protected function shouldCopyFileOnDisk(RemoteFile $file, Media $media, string $diskDriverName): bool
+    {
+        return $file->getDisk() === $media->disk && ($diskDriverName !== 's3' || count($media->getCustomHeaders()) === 0);
```

Comment:
This kind of conditional is unreadable to me. Could you split this up and use early returns?

Reply:
Done, please let me know if that helps. If not I'll be more than happy to improve it further.

For your reference, the conditions I want to meet are:
1. It only affects if the disks are the same.
2. It only affects S3 as far as I know.
3. And only when custom headers are involved which were not present on the source file, to begin with.

Reply:
Thanks!